File: marketing/marketing-batch/src/main/scala/com/talkingdata/marketing/util/JsonUtil.scala
package com.talkingdata.marketing.util

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import net.minidev.json.JSONObject
import net.minidev.json.parser.JSONParser

import scala.collection.JavaConversions.{mapAsScalaMap, mutableMapAsJavaMap}
import scala.collection.mutable


object JsonUtil extends App {

  /**
    * 将map转为json
    *
    * @param map 输入格式 mutable.Map[String,Object]
    * @return
    **/
  def map2Json(map: mutable.Map[String, Object]): String = {

    val jsonString = JSONObject.toJSONString(map)

    jsonString
  }

  /**
    * 将json转化为Map
    *
    * @param json 输入json字符串
    * @return
    **/
  def json2Map(json: String): mutable.HashMap[String, Object] = {

    val map: mutable.HashMap[String, Object] = mutable.HashMap()

    val jsonParser = new JSONParser()

    //将string转化为jsonObject
    val jsonObj: JSONObject = jsonParser.parse(json).asInstanceOf[JSONObject]

    //获取所有键
    val jsonKey = jsonObj.keySet()

    val iter = jsonKey.iterator()

    while (iter.hasNext) {
      val field = iter.next()
      val value = jsonObj.get(field).toString

      if (value.startsWith("{") && value.endsWith("}")) {
        val value = mapAsScalaMap(jsonObj.get(field).asInstanceOf[java.util.HashMap[String, String]])
        map.put(field, value)
      } else {
        map.put(field, value)
      }
    }
    map
  }
}

File: marketing/marketing-streaming/src/main/java/com/talkingdata/marketing/streaming/util/JsonUtil.java
package com.talkingdata.marketing.streaming.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

/**
 * JSON 工具类
 * @author sheng.hong
 */
public class JsonUtil {

    private static final ObjectMapper MAPPER = new ObjectMapper();

    /**
     * 转为JSON串
     * @return
     */
    public static String toJsonString(Object obj) throws JsonProcessingException {
        return MAPPER.writeValueAsString(obj);
    }

    /**
     * 解析JSON串
     * @param jsonString
     * @return
     */
    public static <T> T toObject(String jsonString, Class<T> t) throws IOException {
        return MAPPER.readValue(jsonString, t);
    }
}
